src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,37 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate from = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate to = LocalDate.parse(dateTo, FORMATTER);
+        String[] info = new String[data.length * 4];
+        int c = 0;
+        for (String d : data) {
+            String[] one = d.split(" ");
+            info[c] = one[0];
+            info[c + 1] = one[1];
+            info[c + 2] = one[2];
+            info[c + 3] = one [3];
+            c += 4;
```

Comment:
this can be done more reliably with simple `for i` loop. but, can we manage to solve this using only `one` array (but name it properly:) ) ?